87.md

```diff
@@ -0,0 +1,126 @@
+NIP-87
+======
+
+Ecash Mint Discoverability
+--------------------------------
+
+`draft` `optional`
+
+This NIP describes `kind:38173` and `kind:18173`: a way to discover ecash mints, their capabilities, and people who recommend them.
+
+## Rationale
+
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for users to discover ecash mints recommended by other users and to interact with them.
+
+### Parties involved
+
+There are three actors to this workflow:
+
+* An ecash mint operator, announces their mint and its capabilities.
+    * Publishes `kind:38173`, detailing how to connect to it and its capabilities.
+* user A, who recommends an app that handles a specific event kind
+    * Publishes `kind:18173`
+* user B, who seeks a recommendation for an ecash mint
+    * Queries for `kind:18173` and, based on results, queries for `kind:38173`
+
+## Events
+
+### Recommendation event
+```json
+{
+  "kind": 18173,
+  "pubkey": <recommender-user-pubkey>,
+  "tags": [
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["u", <recommended-fedimint-invite-code>, "fedimint"],
+    ["u", <recommended-cashu-mint-url>, "cashu"]
+  ]
+}
+```
+
+Multiple `u` tags can appear on the same `kind:18173`.
+
+The second value of the tag SHOULD be the string used to connect to the ecash mint.
+The third value of the tag SHOULD be the type of ecash mint (currently `cashu` or `fedimint`).
+
+## Ecash Mint Information
+
+Cashu mints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
+
+For cashu mints, the `u` tag SHOULD be the URL to the cashu mint and it should list the nuts that the cashu mint supports.
+
+```json
+{
+  "kind": 38173,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <random-id>],
+    ["u", "https://cashu.example.com", "cashu"],
+    ["nuts", "1,2,3,4,5,6,7"]
+  ]
+}
+```
+
+Fedimints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
```

Comment:
There's nothing here that really connects the mint broadcast event to the recommendation notes, besides one of the invite strings. I don't see any purpose for these mint events if it's not even recommended that it should be queried in the first place and nothing verifiably links to them.

Reply:
The only real way to link them would be to tag the event id, however, then you cannot update the mint's event if anything changes (new features, fees, etc)

Reply:
The reason I did it this way so that you can recommend a mint without it announcing itself. Also it saves on the required number of queries to be able to find a mint.

However you do bring up a good point of being able to point to the mint's event. Kept the `u` tag stuff but also added a NIP-89 style `a` tag to be able to recommend based on the mint's event.

Reply:
> The reason I did it this way so that you can recommend a mint without it announcing itself

If you want recommendations for private mints you should really consider not using invite codes since they always contain the necessary information to join a federation. https://github.com/nostr-protocol/nips/pull/1110/files#r1524194996